### DISCUSSION TITLE
View not found when run php artisan twill:blocks

### DISCUSSION THREAD

**Discussion by azernov** [area17/twill#2399] [Q&A]
I've tried to add custom form field as it described in documentation. But when I run `php artisan twill:blocks` I get error:
```
Starting to scan block views directory...

   InvalidArgumentException 

  View [twill.blocks..HERE_MY_FULL_PATH_TO_RESOURCES_FOLDER.resources.views.twill.blocks.markdown] not found.

  at vendor/laravel/framework/src/Illuminate/View/FileViewFinder.php:137
    133▕                 }
    134▕             }
    135▕         }
    136▕ 
  ➜ 137▕         throw new InvalidArgumentException("View [{$name}] not found.");
    138▕     }
    139▕ 
    140▕     /**
    141▕      * Get an array of possible view files.

  i   twill.blocks..HERE_MY_FULL_PATH_TO_RESOURCES_FOLDER.resources.views.twill.blocks.markdown was not found: Are you sure the view exists and is a `.blade.php` file? 

      +3 vendor frames 

  4   [internal]:0
      A17\Twill\Commands\GenerateBlocks::A17\Twill\Commands\{closure}(Object(App\Twill\Block\MarkdownBlock))
      +15 vendor frames 

  20  artisan:37
      Illuminate\Foundation\Console\Kernel::handle(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))


```

I found the problem:

https://github.com/area17/twill/blob/f9a096a6e6b579798f09e2773b33a06a95189a00/src/Commands/GenerateBlocks.php#L83

Perhaps, there should be `$blockName`, but not `$basename`.

**Comment by ifox** [comment]
Hi @azernov, this command is not needed in Twill 2 and 3, it is legacy from when we were generating Vue components from Blade defined blocks, which isn't necessary anymore. Which part of the docs are you following?

**Comment by azernov** [comment]
Hi, Quentin (@ifox)

I've tried different ways according to these guides:
https://twillcms.com/guides/creating_custom_components_form_fields_and_blocks.html#content-customize-and-build
https://twillcms.com/docs/block-editor/development-workflow.html

I understand, there are some parts from previous version (for example, `resources/views/admin/blocks` should be `resources/views/twill/blocks`). I try to make Markdown block (as custom Vue component) with toast ui editor.

I prepare BlockMarkdown.vue file by `php artisan twill:blocks. Without this command when I add new block to page (in browser at cms page) it appears with empty content part.

**Comment by azernov** [comment]
It seems, i got it ) I skipped the following part 😄 :


> The naming convention for custom blocks Vue component is deferred from the block's component name. For example, if your block's component name is a17-block-quote, the custom blocks should be assets/js/blocks/BlockQuote.vue. For component name with underscores, for example a17-amazing_quote, it would be assets/js/blocks/BlockAmazing_quote.vue.


So, if I understood correctly, I have to make BlockMarkdown.vue manually in `resources/assets/js/blocks`?
But if I want to bootstrap it first time, it's better to use command `php artisan twill:blocks` which raise error I showed above.
